expyfun/tests/test_experiment_controller.py

```diff
@@ -52,6 +53,30 @@ def test_unit_conversions(hide_window, ws):
     pytest.raises(RuntimeError, ec._convert_units, verts[0], 'deg', 'pix')
 
 
+def test_validate_audio(hide_window):
+    """Test that validate_audio can pass through samples."""
+    with ExperimentController(*std_args, suppress_resamp=True,
+                              **std_kwargs) as ec:
+        ec.set_stim_db(_get_dev_db(ec.audio_type) - 40)  # 0.01 RMS
+        assert ec._stim_scaler == 1.
+        for shape in ((1000,), (1, 1000), (2, 1000)):
+            samples_in = np.zeros(shape)
+            samples_out = ec._validate_audio(samples_in)
+            assert samples_out.shape == (1000, 2)
+            assert samples_out.dtype == np.float32
+            assert samples_out is not samples_in
+        for order in 'CF':
+            samples_in = np.zeros((2, 1000), dtype=np.float32, order=order)
+            samples_out = ec._validate_audio(samples_in)
+            assert samples_out.shape == samples_in.shape[::-1]
+            assert samples_out.dtype == np.float32
+            # transpose that we do will not copy just make a view
+            assert samples_out.base is samples_in
+            samples_out.flags['WRITEABLE'] = True  # undo our read-only safety
+            samples_out[:] = 1
+            assert_allclose(samples_in.T, samples_out)
```

Comment:
I don't get why this passes; `samples_in` is all zeros, and you just changed `samples_out` to all ones.  Are you just proving a point that `samples_out` is a view?  Why do we need to test that?

Reply:
> Are you just proving a point that samples_out is a view? Why do we need to test that?

Yes exactly. The idea is that, if copies are slow, we should keep the number of copies we make to a minimum. But I think the `samples_out.base is samples_in` should already ensure this, so I'll remove these lines.